### DISCUSSION TITLE
Configurable corner for hidden windows

### DISCUSSION THREAD

**Discussion by dantswain** [nikitabobko/AeroSpace#795] [feature-ideas]
Hi!  I think it would be really handy to be able to configure which corner Aerospace tries to use for hidden windows.  I've read the [note on proper monitor arrangement](https://nikitabobko.github.io/AeroSpace/guide#proper-monitor-arrangement) and understand that there are limitations that this is designed to deal with; I'm just wondering if it's possible to add a configuration option to Aerospace that would cause it to try to use, say, the upper right corner instead of the lower right corner?  

My monitors are arranged such that I have two identical standalone displays mounted to my desk with my laptop display on the desk to the bottom right of the right-most display (see screenshot).  Whenever I change which workspace is active in the middle monitor, the hidden windows from the other workspaces get "hidden" into the bottom half of my laptop monitor because it's situated on the bottom right.  It would be a lot of work for me to rearrange my desk to put the monitor on the left.  I can sort of work around it by shifting the arrangement so that the laptop display is pushed up, but then mouse movement between displays feels incongruous.  I could get a stand for my laptop but I'd prefer not to.

It seems like this would be a fairly common physical configuration, so maybe worth investing some time into making this an option?

Thanks for considering!

![image](https://github.com/user-attachments/assets/c894a243-9ca8-46b1-92a0-19e749730ef1)

**Comment by nikitabobko** [comment]
You are mixing several problems into one question

1. It's not possible to use upper corners - it's a macOS limitation
2. There is no need in providing a configuration option for choosing a corner, since it's possible to automatically understand what corner is the most optimal for each monitor

**Reply by dantswain** [reply]
Oh I see. That makes sense, especially (1) as I didn't realize it was a macOS limitation.

Is there maybe an option to push the window to the bottom of the screen rather than the corner?  I can definitely understand that the default desire is to hide as much of the window as possible, but in my case I'd prefer a thin line at the bottom of the screen over having it obscure half of my laptop screen.

To be clear, overall I'm enjoying AeroSpace and consider this to be a small detail; I just thought it worth asking about the possibility and didn't see an existing thread on it (I did search!).

**Reply by jordan-da** [reply]
> 1. It's not possible to use upper corners - it's a macOS limitation

@nikitabobko is this still true? It has been a few years, and macos versions, maybe things are better now? 

I am not a swift developer but I did ask AI if it can code this for me and it seemed to have a solution? Or at the very least it didn't immediately tell me it wasn't possible.

**Comment by JGSphaela** [comment]
Hi, I have encounter the same issue. My monitors setup is quite similar to @dantswain but I have one on it's side. So when I switch workspace on the main monitor the hidden windows will always display on the vertical one. I also didn't realized MacOS have such limitation that windows can not place outside upper corners.

What comes to my mind is, is it possible to just place all the hidden windows at the furthest corner that won't overlap with any monitor? In my case it would means placing all the currently hidden windows on either the lower right corner of the vertical monitor, or lower left corner of the small display(the built-in one). If nothing works, it might be just better hide the windows directly below the display, so it won't overlap with display on the side. It won't works with vertical stacked display tho.

<img width="327" alt="スクリーンショット 2025-05-23 15 34 43" src="https://github.com/user-attachments/assets/16a0e0f2-953b-4c01-bf05-0d3a1c89be15" />

**Reply by jordan-da** [reply]
Well maybe the point is that there _might_ be a free bottom corner at least one of the displays.. in  your case the two outside monitors have free corners. Maybe there is a technical reason aerospace can't use corners outside of the main monitor?
